src/utils/text.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Partition a text to two parts.
+ *
+ * Returns a tuple with a first text of length <= `max_length` and a second text with the rest of the text.
+ * The `pattern` is used to search for the end of the first part.
+ * @param text the text input
+ * @param max_length the max length of the first part
+ * @param pattern the pattern to search.
+ * @returns a tuple of the splited text
+ */
+export function partition_text(text: string, max_length: number, pattern: RegExp): [string, string] {
+    let lastPoint = text.length;
+    do {
+        lastPoint = text.slice(0, lastPoint).search(pattern);
```

Comment:
just thinking
can we simply split the text at max_length, then rfind the regex ? :thinking: